content/en/docs/languages/go/instrumentation.md

```diff
@@ -169,18 +169,39 @@ are useful for aggregating, filtering, and grouping traces. Attributes can be
 added at span creation, or at any other time during the lifecycle of a span
 before it has completed.
 
+Generally speaking, you should try to set attributes during span creation.
+This will incur less overhead compared to setting attributes after
```

Comment:
Isn't that also the official recommendation of the API spec as well? Maybe this should be repeated in all languages doc.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span-creation

_The API documentation MUST state that adding attributes at span creation is preferred to calling SetAttribute later, as samplers can only consider information already present during span creation._

Reply:
good point, do you mind turning your comment in a new issue such that we can track it (and after this PR is merged we may turn it to a shortcode across all languages to be used for that)

Reply:
https://github.com/open-telemetry/opentelemetry.io/issues/3947